scripts/data-fetch.js

```diff
@@ -1,19 +1,43 @@
-var fs = require('fs')
-var request = require('request')
-
-const DATA_URL = 'http://data.nashville.gov/api/views/4ibi-mxs4'
+const fs = require('fs');
+const fetch = require('node-fetch');
+const https = require('https');
 
 /**
- * HTTP requests for latest published General Government Employees Demographics dataset.
+ * Fetches all of the data files stored on the Code For Nashville
+ * Open Data Portal on GitHub and saves the files
+ * in the input directory locally.
+ * These files are needed for the preprartion that will
+ * take place in the data-import script.
  */
-function fetchPublishedData () {
-  request(DATA_URL, function (error, response, body) {
-    if (error) console.error(error)
-    var data = JSON.parse(body)
-    var rowsUpdatedAt = new Date(data['rowsUpdatedAt'] * 1000)
-    var filename = rowsUpdatedAt.toISOString().replace(/-/g, '').slice(0, 8)
-    request(`${DATA_URL}/rows.csv`).pipe(fs.createWriteStream(`./input/${filename}.csv`))
-  })
-}
 
-fetchPublishedData()
+const fetchPublishedData = async () => {
+    const httpsAgent = https.Agent({
+        rejectUnauthorized: false,
+        requestCert: false
+    });
+    
+    const options = {
+        agent: httpsAgent
+    }
+        ,apiEndpoint = 'https://api.github.com/repos'
+        ,repoAddress = '/code-for-nashville/open-data-portal'
+        ,path = '/nashville/metro-general-government-employees-demographic-data/renamed-csv';
+    
+    let res = await fetch(apiEndpoint + repoAddress + '/contents' + path, options);
+    let resolved = await res.ok;
+    if (resolved) {
+        let fileList = await res.json();
+        fileList.map(async x => {
+            let rows = await fetch(x.download_url, options);
+            let data = await rows.text();
+            var filename = x.name.replace(/-/g, '').slice(0, 6) + '01';
```

Comment:
Why are you normalizing to the first of the month? Does some other code expect it or something?

Reply:
Yes, the data-import script, that reads these files, and then creates the data files that the website actually uses for the charts, expects a day of the month.  For the files we are storing on the open data portal, we have just been using month and year (since the data is quarterly, the day doesn't serve much purpose).  The selection of the first day of the month was my arbitrary choice - just going on the basis that every month always has a first day.

Reply:
Yeah - that makes sense. I didn't realize until later that the open data portal doesn't have day of month.